src/mono/mono/metadata/sgen-new-bridge.c

```diff
@@ -654,6 +654,7 @@ static void
 reset_data (void)
 {
 	dyn_array_ptr_empty (&registered_bridges);
+	free_data ();
```

Comment:
I am not sure [yet] why is this needed. There are cases where the data were not freed and `processing_stw_step` started with non-empty `hash_table` still referencing objects from the previous pass. It's not immediately obvious whether this happens only for the bridge comparison tests or whether it can happen for real-life scenarios too.

Reply:
Tarjan bridge frees in `processing_after_callback`. Seems like it makes more sense to have the cleanup there for consistency. But I'm also curious to understand how we come to this.

Reply:
There's actually a very reasonable explanation in the comments:

https://github.com/dotnet/runtime/blob/3eef331e5b402cef529d18d5f88e3ad742a9cf3a/src/mono/mono/sgen/sgen-gc.c#L1179-L1191

https://github.com/dotnet/runtime/blob/3eef331e5b402cef529d18d5f88e3ad742a9cf3a/src/mono/mono/sgen/sgen-gc.c#L2671-L2685

This means it's an actual product bug and not just testing bug. It also requires explicit `GC.Collect(1)` call to trigger if I understand it correctly.